harness/determined/cli/dev.py

```diff
@@ -61,6 +68,201 @@ def curl(args: Namespace) -> None:
+            continue
+        name, params = bindings_sig(obj)
+        if not show_unusable:
+            if not can_be_called_via_cli(params):
+                continue
+            params = [p for p in params if is_supported_annotation(p.annotation)]
+        rv.append((name, params))
+
+    rv.sort(key=lambda x: x[0])
+    odict: OrderedDict[str, List[inspect.Parameter]] = collections.OrderedDict()
+    for name, params in rv:
+        odict[name] = params
+    return odict
+
+
+def parse_param_value(param: inspect.Parameter, value: str) -> Any:
+    annot = unwrap_optional(param.annotation)
+
+    if annot is str:
+        return value
+    elif annot is float:
+        return float(value)
+    elif annot is int:
+        return int(value)
+    elif annot is bool:
+        assert value.lower() in ["true", "false"]
+        return value.lower() == "true"
+    try:
+        value = json.loads(value)
+    except json.JSONDecodeError:
+        raise ValueError(f"Invalid JSON for {param.name}")
+
+    return value
+
+
+def parse_args_to_kwargs(args: List[str], params: List[inspect.Parameter]) -> Dict[str, Any]:
+    kwargs: Dict[str, Any] = {}
+    params_d: Dict[str, inspect.Parameter] = {p.name: p for p in params}
+
+    for idx, arg in enumerate(args):
+        key: str = ""
+        value: Any = None
+        if "=" in arg:
+            key, value = arg.split("=", 1)
+        else:
+            key = params[idx].name
+            value = arg
+        param = params_d.get(key)
+        if not param:
+            raise ValueError(f"Unknown argument {key}")
+        value = parse_param_value(param, value)
+        if key in kwargs:
+            raise ValueError(f"Argument {key} specified twice")
+        kwargs[key] = value
+
+    assert len(kwargs) <= len(params), "too many arguments"
+    return kwargs
+
+
+def print_resposne(d: Any) -> None:
+    if d is None:
+        return
+    if hasattr(d, "to_json"):
+        cli.render.print_json(d.to_json())
+    elif inspect.isgenerator(d):
+        for v in d:
+            print_resposne(v)
+    else:
+        print(d)
+
+
+def list_bindings(args: Namespace) -> None:
+    for name, params in get_available_bindings(show_unusable=args.show_unusable).items():
+        print(bindings_sig_str(name, params))
+
+
+@authentication.required
+def call_bindings(args: Namespace) -> None:
+    """
+    support calling some bindings with primitive arguments via the cli
+    """
+    sess = cli.setup_session(args)
+    fn_name: str = args.name
+    fns = get_available_bindings(show_unusable=False)
+    try:
+        fn = getattr(bindings, fn_name)
+    except AttributeError:
+        matches = [n for n in fns.keys() if re.match(f".*{fn_name}.*", n, re.IGNORECASE)]
+        if not matches:
+            raise errors.CliError(f"no such binding: {fn_name}")
+        print(f"found {len(matches)} matches for '{fn_name}': {list(enumerate(matches))}")
+        user_idx = (
+            input(
```

Comment:
Thanks for being opinionated. The interface is a little weird. If it was my own, I'd probably spit out an error message and exit, but you're already using this and you know what works for you so whatever you like is fine. If you do decide to keep it, could you clean up the language a bit to make it more professional software?

Reply:
I agree it's a bit odd, I have thoughts on making it better but it all comes with more code
the way I've been using it is "I know there is an api called similar to 'comparetrial' but I don't know the full name can I just call it without exactly knowing what it was?" and this flow has helped more that erroring out and exiting. IMO If we run into a concrete downside with it (maybe piping?) we can reevaluate. I updated the wording.